src/SQLProvider/Utils.fs

```diff
@@ -55,17 +55,19 @@ module internal Utilities =
         | [] -> async { () }
 
 
-    let ensureTransaction() =
+    let ensureTransaction (transactionOptions : Transactions.TransactionOptions) =
         // Todo: Take TransactionScopeAsyncFlowOption into use when implemented in Mono.
         // Without it, transactions are not thread-safe over threads e.g. using async can be dangerous)
         // However, default option for TransactionScopeOption is Required, so you can create top level transaction
         // and this Mono-transaction will have its properties.
+        let transactionScopeOption = Transactions.TransactionScopeOption.Required
+
         let isMono = Type.GetType ("Mono.Runtime") <> null
         match isMono with
-        | true -> new Transactions.TransactionScope()
+        | true -> new Transactions.TransactionScope(transactionScopeOption, transactionOptions)
```

Comment:
Are you talking about the code which was there before @Thorium 's commit, which used the reflection? I understood it is there because if we're not in Mono (isMono = false) then we want to use the feature available in .NET, but we have to use reflection because that enum doesn't even exist in Mono and it wouldn't compile.